src/hooks/useApiKeys.ts

```diff
@@ -0,0 +1,199 @@
+import { useState, useCallback } from "react";
+import { useAuth } from "../components/auth/AuthProvider.js";
+
+export enum Scope {
+  RuntRead = "runt:read",
+  RuntExecute = "runt:execute",
+}
+
+export type Resource = {
+  id: string;
+  type: string;
+};
+
+export type CreateApiKeyRequest = {
+  scopes: Scope[];
+  resources?: Resource[];
+  expiresAt: string;
+  name?: string;
+  userGenerated: boolean;
+};
+
+export type ApiKey = CreateApiKeyRequest & {
+  id: string;
+  userId: string;
+  revoked: boolean;
+};
+
+export type ListApiKeysRequest = {
+  limit?: number;
+  offset?: number;
+};
+
+const API_BASE = "/api/api-keys";
+
+export function useApiKeys() {
+  const { accessToken } = useAuth();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const makeAuthenticatedRequest = useCallback(
+    async (url: string, options: RequestInit = {}) => {
+      const token = accessToken;
+      if (!token) {
+        throw new Error("No authentication token available");
+      }
+
+      const response = await fetch(url, {
+        ...options,
+        headers: {
+          ...options.headers,
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json().catch(() => ({}));
+        throw new Error(
+          errorData.message || `HTTP ${response.status}: ${response.statusText}`
+        );
+      }
+
+      if (response.status === 204) {
+        return null;
+      }
+
+      return response.json();
+    },
+    [accessToken]
+  );
+
+  const createApiKey = useCallback(
+    async (request: CreateApiKeyRequest): Promise<string> => {
+      setLoading(true);
+      setError(null);
+
+      try {
+        const response = await makeAuthenticatedRequest(API_BASE, {
+          method: "POST",
+          body: JSON.stringify(request),
+        });
+
+        return response.api_key;
+      } catch (err) {
+        const errorMessage =
+          err instanceof Error ? err.message : "Failed to create API key";
```

Comment:
This is where can start to generate better patterns around structured error handling on the frontend. Please don't worry about it this PR though - just something to build on in the future